catalog/imaging/imaging-plugin-nitf/src/main/java/org/codice/alliance/plugin/nitf/NitfPostProcessPlugin.java

```diff
@@ -291,56 +337,57 @@ private void addThumbnailToMetacard(Metacard metacard, BufferedImage bufferedIma
     }
   }
 
-  private ContentItem createDerivedImage(
-      String id,
+  private ProcessResource createDerivedImage(
       String qualifier,
       BufferedImage image,
       Metacard metacard,
       int maxWidth,
       int maxHeight) {
     try {
       byte[] overviewBytes = scaleImage(image, maxWidth, maxHeight);
+      InputStream overviewBytesInputStream = new ByteArrayInputStream(overviewBytes);
 
-      ByteSource source = ByteSource.wrap(overviewBytes);
-      ContentItem contentItem =
-          new ContentItemImpl(
-              id,
-              qualifier,
-              source,
+     ProcessResource  processResource =
+          new ProcessResourceImpl(
+              metacard.getId(),
+              overviewBytesInputStream,
               IMAGE_JPEG,
               buildDerivedImageTitle(metacard.getTitle(), qualifier, JPG),
               overviewBytes.length,
-              metacard);
+              qualifier);
 
-      addDerivedResourceAttribute(metacard, contentItem);
+      ((ProcessResourceImpl) processResource).markAsModified();
```

Comment:
❓Is there a way we could avoid the cast here? For instance, could we pass the "modified" flag in the constructor or add the method to the `ProcessResource` interface? Not sure about the impact of doing the latter though so we would need to be careful.

Reply:
So far I haven't found a way to avoid it, the constructor doesn't currently accept an argument for modified either 😞

Reply:
I can't remember why it wasn't put it in the interface or ctor, but I think there might have been a reason. Could also be an oversight. @emmberk, does that ring a bell?